rust/operator-binary/src/spark_k8s_controller.rs

```diff
@@ -594,6 +612,31 @@ fn pod_template(
         .resources(config.resources.clone().into())
         .image_from_product_image(spark_image);
 
+    // These env variables enable the `containerdebug` process in driver and executor pods.
+    // More precisely, this process runs in the background of every `spark` container.
+    // - `CONTAINERDEBUG_LOG_DIRECTORY` - is the location where tracing information from the process
+    // is written. This directory is created by the process it's self.
+    // - `_STACKABLE_PRE_HOOK` - is evaluated by the entrypoint script (run-spark.sh) in the Spark images
```

Comment:
Is this a new mechanism? Have we been telling customers to use it for other stuff? That would probably conflict with this.

Reply:
It's been introduced with the logging epic long ago. But it's not advertised anywhere as far as I see probably due to it being an implementation detail.

But maybe @siegfriedweber knows more.

Reply:
The post hook is used for shutting down the Vector agent. The pre hook is not used yet, so it is okay to add the `containerdebug` command.

It is not intended to be used by customers, therefore it is unmistakably prefixed with an underscore. :grin: